README.md

```diff
@@ -1,315 +1,248 @@
-2. Select gear icon next to **Project Overview**
-3. Choose **Service Account** on header
-4. Under **Firebase Admin SDK** section, click on **Generate new private key**
-5. Move the service account credentials under `src/main/resources/<CREDENTIALS_FILENAME>.json`
-6. Update `application.frontend.firebase.service-account-credentials-path` to the filename
-7. Under **Project Overview** page in the **General** tab, find the Firebase config. Update the application properties accordingly.
-8. Create a Firebase Realtime Database and configure the rules as such:
-
-```
-{
-  "rules": {
-    ".read": "auth !== null && auth.token.firebaseAuthorizedUser === true",
-    ".write": "auth !== null && auth.token.firebaseAuthorizedUser === true"
-  }
-}
-```
-
-### 3. Run
-
-```
-
-./mvnw
-
-```
+# OncoKB Curation
+
+## Local Development
+
+1. Before you can build this project, you must install the following
+   dependencies on your machine:
+
+   1. [Docker](https://www.docker.com/get-started).
+   2. [mysql](https://dev.mysql.com/doc/mysql-installation-excerpt/5.7/en/).
+   3. [Java 11](https://dev.java/download/) SDK.
+   4. [Node.js](https://nodejs.org): We use Node to run a development web server
+      and build the project. Depending on your system, you can install Node either
+      from source or as a pre-packaged bundle. The version of node we are using can
+      be found in the [.nvmrc](.nvmrc) file.
+
+      - If you are using [nvm](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating)
+        simply run the following commands from the root directory of the project
+        to install and use the correct version of node.
+
+      ```sh
+      nvm install
+      nvm use
+      ```
+
+   5. [Yarn](https://classic.yarnpkg.com/lang/en/docs/install/)
+
+2. After installing Node and yarn, you should be able to run the following command
+   to install development tools. You will only need to run this command when
+   dependencies change in [package.json](package.json).
+
+   ```sh
+   yarn install
+   ```
+
+   We use yarn scripts and [Webpack](https://webpack.js.org/) as our build system.
+
+3. If you are using redis as a cache, you will have to launch a cache server.
+
+   - To start your cache server, run:
+
+     ```sh
+     docker-compose -f src/main/docker/redis.yml up
+     ```
+
+   - The cache can also be turned off by adding to the [application-dev.yaml](src/main/resources/config/application-dev.yml):
+
+     ```yaml
+     spring:
+       cache:
+         type: none
+     ```
+
+   See [here](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-caching.html#boot-features-caching-provider-none)
+   for details.
+
+4. Configure Keycloak and Google SSO
+
+   To log in to your app, you'll need to have [Keycloak](https://keycloak.org)
+   up and running.
+
+   ```sh
+   docker-compose -f src/main/docker/keycloak.yml up
+   ```
```

Comment:
We do have a script that we can use to bring up/down docker services:
`yarn run docker:keycloak:up`